packages/aws-cdk-lib/aws-cloudfront-origins/test/function-url-origin.test.ts

```diff
@@ -41,3 +46,416 @@ test('Correctly renders the origin for a Lambda Function URL', () => {
+        }),
+      },
+    });
+
+    const template = Template.fromStack(stack);
+
+    // Validate the signing behavior is correctly set to SIGV4_NO_OVERRIDE
+    template.hasResourceProperties('AWS::CloudFront::OriginAccessControl', {
+      OriginAccessControlConfig: {
+        Name: 'CustomLambdaOAC',
+        OriginAccessControlOriginType: 'lambda',
+        SigningBehavior: 'no-override',
+        SigningProtocol: 'sigv4',
+      },
+    });
+  });
+
+  test('Correctly adds permission for an imported Lambda Function', () => {
+    const importedFn = lambda.Function.fromFunctionArn(stack, 'ImportedFunction', 'arn:aws:lambda:us-east-1:123456789012:function:my-imported-fn');
+
+    const fnUrl = importedFn.addFunctionUrl({
+      authType: lambda.FunctionUrlAuthType.AWS_IAM,
+    });
+
+    new cloudfront.Distribution(stack, 'MyDistribution', {
+      defaultBehavior: {
+        origin: FunctionUrlOrigin.withOriginAccessControl(fnUrl, {}),
+      },
+    });
+
+    const template = Template.fromStack(stack);
+
+    template.hasResourceProperties('AWS::Lambda::Permission', {
+      Action: 'lambda:InvokeFunctionUrl',
+      FunctionName: {
+        'Fn::GetAtt': ['ImportedFunctionFunctionUrlB3FF8A17', 'FunctionArn'],
+      },
+      Principal: 'cloudfront.amazonaws.com',
+      SourceArn: {
+        'Fn::Join': [
+          '',
+          [
+            'arn:',
+            { Ref: 'AWS::Partition' },
+            ':cloudfront::',
+            { Ref: 'AWS::AccountId' },
+            ':distribution/',
+            { Ref: 'MyDistribution6271DFB5' },
+          ],
+        ],
+      },
+    });
+  });
+  test('Correctly creates a Lambda Function URL Origin with default properties', () => {
+    const fn = new lambda.Function(stack, 'MyFunction', {
+      code: lambda.Code.fromInline('exports.handler = async () => {};'),
+      handler: 'index.handler',
+      runtime: lambda.Runtime.NODEJS_20_X,
+    });
+
+    const fnUrl = fn.addFunctionUrl({
+      authType: lambda.FunctionUrlAuthType.AWS_IAM,
+    });
+
+    const origin = FunctionUrlOrigin.withOriginAccessControl(fnUrl);
+
+    const originBindConfig = origin.bind(stack, { originId: 'StackOriginLambdaFunctionURL' });
+
+    expect(stack.resolve(originBindConfig.originProperty)).toEqual({
+      id: 'StackOriginLambdaFunctionURL',
+      domainName: {
+        'Fn::Select': [
+          2,
+          {
+            'Fn::Split': [
+              '/',
+              { 'Fn::GetAtt': ['MyFunctionFunctionUrlFF6DE78C', 'FunctionUrl'] },
+            ],
+          },
+        ],
+      },
+      customOriginConfig: {
+        originProtocolPolicy: 'https-only',
+        originSslProtocols: ['TLSv1.2'],
+      },
+      originAccessControlId: {
+        'Fn::GetAtt': [
+          'FunctionUrlOriginAccessControlC9E60518',
+          'Id',
+        ],
+      },
+    });
+  });
+  test('Creates Lambda Function URL origin with default OAC and origin properties', () => {
```

Comment:
nit: seems this test is duplicated. It is similar to test `Correctly creates a Lambda Function URL Origin with default properties`

Reply:
https://github.com/aws/aws-cdk/pull/31339/commits/2375bcd40c2e7512d1cf0cad40dd6bf98b09d3a7